### ISSUE TITLE
Add 3D discretization of polygons

### ISSUE THREAD

**Issue by juliohm** [JuliaGeometry/Meshes.jl#218]
Our current discretization methods (e.g. FIST, Dehn1989) are defined for 2D `Polygon`. We can easily generalize these methods to 3D with the following steps:

1. Project 3D `Polygon` onto a 3D `Plane`
2. Rotate `Plane` to a horizontal 2D `Plane`
3. Perform discretization method
4. Return `SimpleMesh` with original vertices

**Comment by juliohm** [comment]
This is already implemented for polygons without holes. We can take any polygon with 3D coordinates and discretize it with any  discretization method. The problem with holes is that we need to build bridges between the outer and inner chains of the polygon, and this algorithm needs to be refactored before we can consider the 3D case. For most practical applications, users won't need 3D discretization of polygons with holes. These polygons usually appear in GIS where polygons are represented with 2D lat/lon coordinates anyways.

**Comment by juliohm** [comment]
Exotic examples where "invalid" polygons with 3D coordinates just work:

```julia
using Meshes, MeshViz

import CairoMakie as Mke

# this octagon is invalid because its vertices don't live in a common
# plane, however we can still triangulate it using a SVD projection
o = Octagon([(0.0,0.0,1.0), (0.5,-0.5,0.0), (1.0,0.0,0.0), (1.5,0.5,-0.5),
             (1.0,1.0,0.0), (0.5,1.5,0.0), (0.0,1.0,0.0), (-0.5,0.5,0.0)])
viz(o)
```
![image](https://user-images.githubusercontent.com/3345261/145627498-06ae4357-9ee2-47d5-beb7-ace6189bc961.png)


```julia
viz(o, showboundary = false)
```
![image](https://user-images.githubusercontent.com/3345261/145627559-055c7f96-935a-4f18-b358-0c07f710aae2.png)


```julia
viz(o, color = :orange)
```
![image](https://user-images.githubusercontent.com/3345261/145627625-6a11d82e-ebd5-4455-8e02-f98678d4ff2b.png)

```julia
viz(o, color = :cyan, boundarycolor = :red)
```
![image](https://user-images.githubusercontent.com/3345261/145627768-e145475e-8dc6-487d-b45b-fbed7f80a76c.png)

Here is the underlying mesh:

```julia
viz(triangulate(o), showfacets = true)
```
![image](https://user-images.githubusercontent.com/3345261/145628051-a14cc50c-5704-4217-8023-984e6b73c687.png)

**Comment by juliohm** [comment]
I will mark this issue as solved. We can open a new issue if the discretization of polygons with holes embedded in R^3 becomes important.
